scipy/stats/_entropy.py

```diff
@@ -135,6 +147,26 @@ def differential_entropy(
     the (unknown) distribution. In general, the smoother the density of the
     distribution, the larger is such optimal value of ``window_length`` [1]_.
 
+    The following options are available for the `method` parameter.
+
+    * ``'vasicek'`` uses the estimator presented in [1]. This is
+      one of the first and most influential estimators of differential entropy.
+    * ``'van es'`` uses the bias-corrected estimator presented in [3], which is
+      not only consistent but, under some conditions, asymptotically normal.
+    * ``'ebrahimi'`` uses an estimator presented in [4], which was shown
+      in simulation to have smaller bias and mean squared error than
+      the Vasicek estimator.
+    * ``'correa'`` uses the estimator presented in [5] based on local linear
+      regression. In a simulation study, it had consistently smaller mean
+      square error than the Vasiceck estimator, but it is more expensive to
+      compute.
+    * ``'auto'`` selects the method automatically (default). Currently,
+      this selects ``'van es'`` for very small samples (<10), ``'ebrahimi'``
+      for moderate sample sizes (11-1000), and ``'vasicek'`` for larger
```

Comment:
Is there a reason why you didn't default to `None` for the method parameter? That way you can say `"if method is not specified, then ....". This has the advantage that "auto" does not appear in the exception message when a user passes the wrong string since it is the default option.

Reply:
Considered `None`, but I guess I went with how methods are selected in some existing statistical tests, e.g. [`kstest`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kstest.html) and some new ones being added in 1.7 (e.g. [`pagel`](http://scipy.github.io/devdocs/reference/generated/scipy.stats.page_trend_test.html), [`cramervonmises_2samp`](http://scipy.github.io/devdocs/reference/generated/scipy.stats.cramervonmises_2samp.html)). I'll think about it some more, thanks!

Reply:
I've thought about it a bit and if we didn't want it to appear in the exception message, I can see the advantage. But I don't really see why it shouldn't appear in the exception message. Either `None` or `auto` is a valid value of the parameter.